src/index.js

```diff
@@ -73,4 +74,26 @@ export default class Banana {
     }
     return messageKey
   }
+
+  /**
+   * Register a plugin for the library's message parser
+   * Example:
+   * <pre>
+   *   banana.registerParserPlugin('sysop', nodes => {
+   *     return mw.config.get('wgUserGroups').includes('sysop') ?
+   *       nodes[1] : nodes[2]
+   *   }
+   * </pre>
+   * This allows messages to know if the user is a sysop. Usage:
+   * <pre>
+   *   banana.i18n('this is a message {{sysop:_|Message for sysops|Message for non-sysops}}')
+   * </pre>
+   * See emitter.js for built-in parser operations.
+   * @param {string} name - the name of the parser hook
```

Comment:
This could be 'name of the plugin'? - Just to make it consistent with naming? 

plugin and pluginname